aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -250,4 +262,160 @@ void gemm(
       transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
 
+DEFINE_DISPATCH(axpy_stub);
+
+void axpy(int64_t n, double a, const double *x, int64_t incx, double *y, int64_t incy) {
+  if(n == 1)
+  {
+    incx = 1;
+    incy = 1;
+  }
+  #if AT_BUILD_WITH_BLAS()
+  if( (n <= INT_MAX) && (incx <= INT_MAX) && (incy <= INT_MAX) )
+  {
+    int i_n = (int)n;
+    int i_incx = (int)incx;
+    int i_incy = (int)incy;
+    daxpy_(&i_n, &a, x, &i_incx, y, &i_incy);
+    return;
+  }
+  #endif 
+  axpy_stub(
```

Comment:
I have hard time believing that naive implementation in axpy_stub is better than y.add_(x, a) (or whatever the right syntax is for inplace add with alpha value set). Same for copy.

Reply:
For small `n` it avoids overhead from tensor creation (for the slice), operator dispatch and `TensorIterator`.

Reply:
There's no tensor creation for inplace ops, and with blas-enabled builds it'll come here for large tensors, not small.

Reply:
>There's no tensor creation for inplace ops

I'm talking about creating the view into the tensor in order to copy/axpy some subset of it. In all cases, that's how the `THBlas` `axpy`/`copy` operations were used.

> with blas-enabled builds it'll come here for large tensors, not small.

1. the naive code is primarily for non-blas types like integers.
2. you can also have small data with large strides and blas can't handle it

Reply:
> > There's no tensor creation for inplace ops
> 
> I'm talking about creating the view into the tensor in order to copy/axpy some subset of it. In all cases, that's how the `THBlas` `axpy`/`copy` operations were used.
> 
> > with blas-enabled builds it'll come here for large tensors, not small.
> 
> 1. the naive code is primarily for non-blas types like integers.
> 2. you can also have small data with large strides and blas can't handle it

I agree with Peter.